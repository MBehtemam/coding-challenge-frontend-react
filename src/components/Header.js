import styled from 'styled-components';

const Header = styled.header`
  background-color: rgb(219, 112, 147);
  color: white;
  font-size: 3rem;
  height: 10vh;
`;
export default Header;
