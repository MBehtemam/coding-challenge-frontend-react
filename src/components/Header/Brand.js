import styled from 'styled-components';

const Brand = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
`;
export default Brand;
