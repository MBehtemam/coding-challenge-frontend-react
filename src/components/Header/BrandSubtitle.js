import styled from 'styled-components';

const BrandSubtitle = styled.h3`
  color: white;
  font-size: 1.5rem;
`;
export default BrandSubtitle;
