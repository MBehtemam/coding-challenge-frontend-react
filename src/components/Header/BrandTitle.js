import React from 'react';
import styled from 'styled-components';

const BrandTitle = styled.h1`
  color: white;
  font-size: 2rem;
`;
export default BrandTitle;
