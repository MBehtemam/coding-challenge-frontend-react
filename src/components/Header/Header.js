import React from 'react';
import styled from 'styled-components';
import Logo from './Logo';
import Brand from './Brand';
import BrandTitle from './BrandTitle';
import BrandSubtitle from './BrandSubtitle';

const Wrapper = styled.header`
  align-items: center;
  background-color: rgb(219, 112, 147);
  color: white;
  display: flex;
  min-height: 90px;
`;

const Header = () => (
  <Wrapper>
    <Logo src={require('../../assets/logo.jpg')} />
    <Brand>
      <BrandTitle>Police Department of Berlin</BrandTitle>
      <BrandSubtitle>Stolen bykes</BrandSubtitle>
    </Brand>
  </Wrapper>
);

export default Header;
