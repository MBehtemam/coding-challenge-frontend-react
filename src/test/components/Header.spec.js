import React from 'react';
import Header from '../../components/Header';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
describe('Tests for Header Component', () => {
  it('Should correctly render header', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should has correct height', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toHaveStyleRule('height', '10vh');
  });
});
