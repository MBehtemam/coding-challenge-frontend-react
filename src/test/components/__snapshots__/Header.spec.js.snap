// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Header Component Should correctly render header 1`] = `
.c0 {
  background-color: rgb(219,112,147);
  color: white;
  font-size: 3rem;
  height: 10vh;
}

<header
  className="c0"
/>
`;
